lightning/src/onion_message/messenger.rs

```diff
@@ -297,6 +319,45 @@ where
 		}
 	}
 
+	fn respond_with_onion_message<T: CustomOnionMessageContents, EH: ResponseErrorHandler>(
+		&self, response: OnionMessageContents<T>, path_id: Option<[u8; 32]>,
+		reply_path: Option<BlindedPath>, error_handler: &EH
+	) {
+		let sender = match self.node_signer.get_node_id(Recipient::Node) {
+			Ok(node_id) => node_id,
+			Err(_) => {
+				log_info!(
+					self.logger, "No sender to respond to onion message with path_id {:02x?}",
+					path_id
+				);
+				return error_handler.handle_response_error(ResponseError::GetNodeIdFailed);
+			}
+		};
+
+		let destination = match reply_path {
+			Some(reply_path) => Destination::BlindedPath(reply_path),
+			None => {
+				log_info!(
+					self.logger, "No reply path to respond to onion message with path_id {:02x?}",
+					path_id
+				);
+				return error_handler.handle_response_error(ResponseError::NoReplyPath);
+			},
+		};
+
+		let intermediate_nodes = self.message_router.find_route(&sender, &destination);
```

Comment:
Should `find_route` return `Option<Vec<Pubkey>>` to better express the case where we can't find a route to the `Destination`?

Assuming `find_route` returns an empty vector if it can't find one, `send_onion_message` will get all the way to returning [InvalidHop](https://github.com/lightningdevkit/rust-lightning/blob/5c89d019050a4b2dd68610d2e59d45acbfee5775/lightning/src/onion_message/messenger.rs#L257) if we're not peered with the `Destination` - seems like the wrong error for that case?

Reply:
Good point! Updated accordingly, though perhaps a `Result<Vec<Pubkey>, ()>` would be more appropriate? Open to hear what others think.